source/features/branch-buttons.tsx

```diff
@@ -38,7 +38,7 @@ async function getTagLink(): Promise<'' | HTMLAnchorElement> {
 		latestRelease = tags[0];
 	}
 
-	const link = <a className="btn btn-sm btn-outline tooltipped tooltipped-ne">{icons.tag()}</a> as HTMLAnchorElement;
+	const link = <a className="btn btn-sm btn-outline tooltipped tooltipped-ne">{icons.tag()}</a> as unknown as HTMLAnchorElement;
```

Comment:
This is the only drawback of this new approach. The element is still internally consistent (`<a>` accepts a `href` attribute but not a `value`) but once out it's a plain `HTMLElement` and we get this error without `as unknown`:


> source/features/branch-buttons.tsx:41:15 - error TS2352: Conversion of type '`Element`' to type '`HTMLAnchorElement`' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
>   Type '`Element`' is missing the following properties from type '`HTMLAnchorElement`': `charset`, `coords`, `download`, `hreflang`, and 20 more.
> 
> 41  `const link = <a className="btn btn-sm btn-outline tooltipped tooltipped-ne">{icons.tag()}</a>  as HTMLAnchorElement;`
>

Reply:
Yeah, I really want to nail down getting these types right for dom-chef. I think dom-chef is the first library to use JSX in a non-React/React like library which is why these issues are popping up.

Reply:
There’s another one called `h` I think but it’s abandoned. I’m sure that there are more as I occasionally find comments of people having similar type issues

Reply:
Yeah, `h` is what Preact uses for vdom.

Reply:
I'm referring to https://www.npmjs.com/package/h, it's dom-based. There's also:

https://github.com/glixlur/jsx-dom
https://github.com/egoist/dom-dom
https://github.com/treycordova/nativejsx (this was promising but it didn't support SVG)
https://github.com/oychao/jsx-dom-render
https://github.com/qwtel/create-element-x

and more: https://npms.io/search?q=dom+jsx

Maybe they have some good ideas we can learn from

Reply:
Yeah using the `unknown` was what I suggested at the beginning but wrapping it in a function to avoid `unknown` everywhere, but that means wrapping all JSX in a function whose main purpose is just to please the TypeScript compiler. https://github.com/sindresorhus/refined-github/issues/1735#issuecomment-462015923

Reply:
All of those new `as unknown` in 9a527dffb45f84742306d3032c110a6d36f7aa34 were not necessary, it was a wrong merge, probably. Only **this one** was necessary.

I sent a new merge